src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,182 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, { useState } from 'react';
+import { Todo } from '../../types/Todo';
+import cn from 'classnames';
+import { USER_ID, deleteTodo, updateTodo } from '../../api/todos';
+
+type TodoItemProps = {
+  todo: Todo;
+  onChangeTodo: (newTodo: Todo) => void;
+  onDeleteTodo: (todoId: number) => void;
+  onErrorMessage: (error: string) => void;
+};
+
+export const TodoItem: React.FC<TodoItemProps> = ({
+  todo,
+  onChangeTodo,
+  onDeleteTodo,
+  onErrorMessage,
+}) => {
+  const { id, completed, title } = todo;
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [todoStatus, setTodoStatus] = useState(completed);
+
+  const [isEditing, setIsEditing] = useState(false);
+  const [newTitle, setNewTitle] = useState(title);
+
+  function handleDeleteTodo(todoId: number) {
+    setIsLoading(true);
+
+    deleteTodo(todoId)
+      .then(() => {
+        onDeleteTodo(todoId);
+
+        if (isEditing) {
+          setIsEditing(false);
+        }
+      })
+      .catch(() => {
+        onErrorMessage('Unable to delete a todo');
+      })
+      .finally(() => setIsLoading(false));
+  }
+
+  function handleChangeTodo(todoId: number, newTodoProp: string) {
```

Comment:
Consider will be better if you use function expression everywhere instead of function declaration, fix it everywhere
```suggestion
  const handleChangeTodo = (todoId: number, newTodoProp: string) => {
```